contracts/MarginBase.sol

```diff
@@ -397,14 +390,21 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
     /// @dev _sizeDelta will always be non-zero
     /// @param _sizeDelta: size and position type (long/short) denominated in market synth
     /// @param _market: synthetix futures market
-    /// @return sizeDeltaInUSD _sizeDelta *in sUSD*
+    /// @return fee *in sUSD*
     function modifyPositionForMarket(int256 _sizeDelta, IFuturesMarket _market)
         internal
-        returns (uint256 sizeDeltaInUSD)
+        returns (uint256 fee)
     {
-        /// @notice _sizeDelta is measured in the underlying base asset of the market
+        // determine trade fee based on size delta
+        fee = calculateTradeFee(_abs(_sizeDelta));
+
+        /// @notice fee is currently measured in the underlying base asset of the market
         /// @dev fee will be measured in sUSD, thus exchange rate is needed
-        sizeDeltaInUSD = (sUSDRate(_market) * _abs(_sizeDelta)) / 1e18;
+        fee = (sUSDRate(_market) * fee) / 1e18;
+
+        /// @notice alter the amount of margin in specific market position
+        /// @dev positive input triggers a deposit; a negative one, a withdrawal
+        _market.transferMargin(int256(fee) * -1);
```

Comment:
could see an edge case popping up here where if you're at max leverage or above you won't be able to modify your position and bring it down because of the fee transfer failiong

consider moving this to after modify position

Reply:
Good point. Changed